src/freenet/node/NewPacketFormat.java

```diff
@@ -10,6 +10,7 @@
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.concurrent.ConcurrentSkipListMap;
```

Comment:
Is there a reason you chose `ConcurrentSkipListMap` here? From what I see, we’re mostly iterating over it so this doesn’t really gain us anything…